src/refactorings/change-signature/change-signature.ts

```diff
@@ -135,13 +144,32 @@ function updateCode(
 
 export function createVisitor(
   selection: Selection,
-  onMatch: (path: t.NodePath<t.FunctionDeclaration>) => void
+  onMatch: (
+    path: t.NodePath<t.FunctionDeclaration | t.ArrowFunctionExpression>,
+    arrowSelection: Selection
+  ) => void
 ): t.Visitor {
   return {
     FunctionDeclaration(path) {
       if (!selection.isInsidePath(path)) return;
 
-      onMatch(path);
+      onMatch(path, selection);
+    },
+    ArrowFunctionExpression(path) {
+      if (!selection.isInsidePath(path)) return;
+
+      if (!t.isVariableDeclarator(path.parent)) return;
+
+      if (!path.parent.loc) return;
+
+      // For code Line always starts at 1. So we need to start at 0.
+      // Starting at 1 will get body function instead of function declaration
+      const arrowSelection = new Selection(
+        [path.parent.loc.start.line - 1, path.parent.loc.start.column],
+        [path.parent.loc.end.line - 1, path.parent.loc.end.column]
+      );
```

Comment:
🍰 I think you should be able to use `Selection.fromAST(path.parent.loc)` for that. It handles the line start at 1 and all, so you can drop the comment in this case.

To be confirmed with tests, but that is a common issue that I solved in the past IMO.

Reply:
Oh! Nice factory.

Used at https://github.com/nicoespeon/abracadabra/commit/d3a78c1001b98783c49aef7234faef6160159ca4
✅ Test OK!